### PR TITLE
More model tests #7

### PR DESCRIPTION
**Pull request by swaterkamp**
Includes a polyfill fix for settings model and tests for:
* Scanner
* ScanConfig
* Credential
* Schedule
* SecInfo
* Settings
* Tag
* Target
* User
* Vulnerability

### CODE REVIEW

File: gsa/src/gmp/models/__tests__/settings.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+import Settings from 'gmp/models/settings';
+
+describe('Settings model tests', () => {
+
+  test('should have private settings', () => {
+    const settings = new Settings({});
+
+    expect(settings._settings).toBeDefined();
```
Comment (bjoernricks) [inline gsa/src/gmp/models/__tests__/settings.js]: Please don't test internal API. Internal API is marked with the underscore. The tests should only ensure the public api is working correctly and the internal api may change every time.

Comment (bjoernricks) [reply gsa/src/gmp/models/__tests__/settings.js]: This concerns all tests in this module.

Comment (swaterkamp) [reply gsa/src/gmp/models/__tests__/settings.js]: Doing this, I could only test set() and get() within a single test, but it should work. With the last push, I also added more commits, so just that you know :)

Comment (bjoernricks) [reply gsa/src/gmp/models/__tests__/settings.js]: > I could only test set() and get() within a single test

That's completely fine.

**Pull request merged**